### ISSUE TITLE
Score label is not functional when we replay the games

### ISSUE THREAD

**Issue by anubhavpulkit** [systers/PowerUp#353]
## Description
When we replay `Vocab Matching Game`, `Minesweeper Game` and `Sink To Swim Game` final score is not updated in the upper left most side.

To update the value of score first we have to make a decision:-
i) we update a new value in total score by removing the old value of that game.
ii) we update the value if the new score is greater then the last score of that game.

## Mocks
![sys issue 789](https://user-images.githubusercontent.com/47811606/75566757-49432780-5a76-11ea-8406-6a90549521bc.png)

### Update [Required]
- [x] [Changes in result code]

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
[6] hours

**Comment by anubhavpulkit** [comment]
@sunjunkie Please have a look can I work on this issue?

**Comment by anubhavsingh16** [comment]
I'm working on this issue.

**Comment by sunjunkie** [comment]
Yes, you can work on this. First check to see what the Android app does about this score.

**Comment by anubhavpulkit** [comment]
@sunjunkie Is it important to use Android algorithm to count final score, Can I propose my own algorithm/idea to count final score?

**Comment by sunjunkie** [comment]
The apps should be consistent. Is there something wrong with the score calculation in the Android version?

**Comment by anubhavpulkit** [comment]
In Android version sequence of mini-games is different than iOS and calculation of score in Android is totally unpredictable I play 8-9 times a full game but I can't understand the logic of there Total score.

**Comment by anubhavpulkit** [comment]
Actually when I tap reply in android Powerup app it opens new mini games so
but in some positions score label work fine.

**Comment by anubhavpulkit** [comment]
@sunjunkie Are you available on this weekend for help?

**Comment by sunjunkie** [comment]
@anubhavsingh16 Yes, I can check in periodically.

Go ahead and decide on an algorithm to update the score. Be sure to document that well in the code and in your PR description. We can always change this algorithm later if we need to.

**Comment by anubhavpulkit** [comment]
Ok, Thank You @sunjunkie

**Comment by anubhavpulkit** [comment]
In the current scenario, we have only one variable which store points is `score` which is not updating after the replay of the game. I want to propose 2 variables currentScore and maxScore.
if the user plays the first time and gain "x" points then we store currentScore and maxScore is equal to x and we use currentScore for score view(which come after game is over) and maxScore for the total score. If user replay the game then, if(currentScore > maxScore) {
then we update the maxScore is equal to currentScore
} else maxScore is remain the last score and currentScore is update and show their current score in score view.

**Comment by anubhavpulkit** [comment]
@what you think about this?

> In the current scenario, we have only one variable which store points is `score` which is not updating after the replay of the game. I want to propose 2 variables currentScore and maxScore.
> if the user plays the first time and gain "x" points then we store currentScore and maxScore is equal to x and we use currentScore for score view(which come after game is over) and maxScore for the total score. If user replay the game then, if(currentScore > maxScore) {
> then we update the maxScore is equal to currentScore
> } else maxScore is remain the last score and currentScore is update and show their current score in score view.

@sunjunkie any suggestions?

**Comment by sunjunkie** [comment]
ok. sounds good.

**Comment by anubhavpulkit** [comment]
@sunjunkie I want to add this issue in my proposal. And want to do it later after the proposal submission. Because I have an idea which is linked with this issue so can you give me permission to take some time and present my proposal, if you like my idea then we work on new method to calculate and use scores otherwise I solve this issue with the simple method which is explain above.

**Comment by anubhavpulkit** [comment]
> ok. sounds good.

Thank You @sunjunkie

**Comment by anubhavpulkit** [comment]
@sunjunkie Please have a look.

**Comment by sunjunkie** [comment]
Have a look at what?

**Comment by anubhavpulkit** [comment]
Sorry @sunjunkie 🙄
